src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -209,6 +210,30 @@ public String getMonitoringEndpoint(Region region) {
     }
 
     private List<List<Dimension>> getDimensions(MetricRule rule, AmazonCloudWatchClient client) {
+        if (
+                rule.awsDimensions != null &&
+                rule.awsDimensions.size() == 1 &&
+                rule.awsDimensionSelect != null &&
+                rule.awsDimensionSelect.size() == 1 &&
+                rule.awsDimensionSelect.containsKey(rule.awsDimensions.get(0))
```

Comment:
Why do we want to limit the size of `awsDimensions` -> For example, consider this scenario I want to fetch metric `SuccessfulRequestLatency`  with dimensions `TableName, Operation` and filter out the metrics based on `TableName` == `my-table`. We can't achieve this with the above functionality.  Basically, what we need here is if there is one `awsDimensionSelect ` value for a particular `awsDimension` then we are good to go with the new way else fall back into old way.

Reply:
Also, add corresponding unit test to test this functionality.

Reply:
@nishanth-pinnapareddy I stuck to only one dimension as that was my current use case.  But I will do a stab at it.

On the unit tests, I will cook up something.